pkg/cmdlinefilter/cmdlinefilter.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package cmdlinefilter provides filters to modify the bootloader kernel
+// parameters before booting.
+package cmdlinefilter
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"unicode"
+
+	"github.com/u-root/u-root/pkg/cmdline"
+)
+
+// Remove filter the given cmdline by removing fields.
+func Remove(cl string, fields []string) string {
+	var newCl []string
+
+	// kernel variables must allow '-' and '_' to be equivalent in variable
+	// names. We will replace dashes with underscores for processing.
+	for i, f := range fields {
+		fields[i] = strings.Replace(f, "-", "_", -1)
+	}
+
+	lastQuote := rune(0)
+	quotedFieldsCheck := func(c rune) bool {
+		switch {
+		case c == lastQuote:
+			lastQuote = rune(0)
+			return false
+		case lastQuote != rune(0):
+			return false
+		case unicode.In(c, unicode.Quotation_Mark):
+			lastQuote = c
+			return false
+		default:
+			return unicode.IsSpace(c)
+		}
+	}
+
+	for _, flag := range strings.FieldsFunc(string(cl), quotedFieldsCheck) {
```

Comment:
This is almost the same code as ParseToMap -- can this be shared somehow?

Reply:
Could you use ParseToMap?

Reply:
I cannot just use `parseToMap` as it potentially store each variables twice. But there still way to share most code.

Reply:
done